### DISCUSSION TITLE
Data retention configuration and auto drop expired partitions

### DISCUSSION THREAD

**Discussion by sunng87** [questdb/questdb#1476] [Ideas]
According to [data retention](https://questdb.io/docs/operations/data-retention) section of the doc, the data retention policy on questdb is implemented as drop-able partitions. While this is flexible to implement custom policy of expiry, I still miss influxdb's straight-forward way: defining data retention policy, delete expired data automatically.  It requires no additional cronjob or operation script to keep the database sustaining. 

The idea is to add a `MAX_PARTITIONS` option to `CREATE TABLE` command:

```sql
CREATE TABLE my_table (timestamp TIMESTAMP, x LONG) timestamp(timestamp)
PARTITION BY DAY MAX_PARTITIONS 30;
```

So that partitions are deleted in a FIFO manner when its count exceeds 30.

**Comment by robolyst** [comment]
Hey, I second this. The biggest missing piece for me in QuestDB right now is an automated retention policy.

**Comment by tucachmo2202** [comment]
I also hope this feature will be implemented too

**Comment by xyzhieu** [comment]
I also hope to have this feature in the future

**Comment by dejurin** [comment]
Must have!

**Comment by nwoolmer** [comment]
TTL has been added: https://questdb.com/docs/reference/sql/alter-table-set-ttl/
